accounts/abi/type.go

```diff
@@ -120,19 +129,52 @@ func NewType(t string, internalType string, components []ArgumentMarshaling) (ty
 
 	// varSize is the size of the variable
 	var varSize int
+	numericTypes := map[string]bool{"int": true, "uint": true, "fixed": true, "ufixed": true}
 	if len(parsedType[3]) > 0 {
+		if !numericTypes[parsedType[1]] && parsedType[1] != "bytes" {
+			return Type{}, fmt.Errorf("only int, uint, fixed, ufixed, bytes types can have variable size specified: %s", t)
+		}
 		var err error
-		varSize, err = strconv.Atoi(parsedType[2])
+		varSize, err = strconv.Atoi(parsedType[3])
 		if err != nil {
 			return Type{}, fmt.Errorf("abi: error parsing variable size: %v", err)
 		}
+		switch {
+		case strconv.Itoa(varSize) != parsedType[3]:
+			fallthrough
+		case varSize == 0:
+			fallthrough
+		case varSize > 256:
+			fallthrough
+		case varSize%8 != 0 && parsedType[1] != "bytes":
+			fallthrough
+		case varSize > 32 && parsedType[1] == "bytes":
+			return Type{}, fmt.Errorf("abi: wrong variable size: %s", parsedType[3])
+		}
```

Comment:
Why did you introduce this error mode? I was wondering why you changed `uint17` to `uint24` in the tests, but clearly it's because the `uint17` causes an error to be thrown here. 
That is breaking backwards compatibility, so unless we have a really good reason to do it I don't see why we should change this?

Reply:
Ah, from Solidity docs: 

https://docs.soliditylang.org/en/latest/types.html

> int / uint: Signed and unsigned integers of various sizes. Keywords uint8 to uint256 in steps of 8 (unsigned of 8 up to 256 bits) and int8 to int256. uint and int are aliases for uint256 and int256, respectively.